polaris-core/src/main/java/org/apache/polaris/core/storage/s3compatible/S3CompatibleCredentialsStorageIntegration.java

```diff
@@ -0,0 +1,122 @@
+      @NotNull PolarisDiagnostics diagnostics,
+      @NotNull S3CompatibleStorageConfigurationInfo storageConfig,
+      boolean allowListOperation,
+      @NotNull Set<String> allowedReadLocations,
+      @NotNull Set<String> allowedWriteLocations) {
+
+    String caI = System.getenv(storageConfig.getS3CredentialsCatalogAccessKeyId());
+    String caS = System.getenv(storageConfig.getS3CredentialsCatalogSecretAccessKey());
+
+    EnumMap<PolarisCredentialProperty, String> propertiesMap =
+        new EnumMap<>(PolarisCredentialProperty.class);
+    propertiesMap.put(PolarisCredentialProperty.AWS_ENDPOINT, storageConfig.getS3Endpoint());
+    propertiesMap.put(
+        PolarisCredentialProperty.AWS_PATH_STYLE_ACCESS,
+        storageConfig.getS3PathStyleAccess().toString());
+    if (storageConfig.getS3Region() != null) {
+      propertiesMap.put(PolarisCredentialProperty.CLIENT_REGION, storageConfig.getS3Region());
+    }
+
+    LOGGER.debug("S3Compatible - createStsClient()");
+    StsClientBuilder stsBuilder = software.amazon.awssdk.services.sts.StsClient.builder();
+    stsBuilder.endpointOverride(URI.create(storageConfig.getS3Endpoint()));
+    if (storageConfig.getS3ProfileName() != null) {
+      stsBuilder.credentialsProvider(
+          ProfileCredentialsProvider.builder()
+              .profileFile(ProfileFileSupplier.defaultSupplier())
+              .profileName(storageConfig.getS3ProfileName())
+              .build());
+      LOGGER.debug("S3Compatible - stsClient using profile from catalog settings");
+    } else if (caI != null && caS != null) {
+      stsBuilder.credentialsProvider(
+          StaticCredentialsProvider.create(AwsBasicCredentials.create(caI, caS)));
+      LOGGER.debug("S3Compatible - stsClient using keys from catalog settings");
+    }
+    try (StsClient stsClient = stsBuilder.build()) {
+      LOGGER.debug("S3Compatible - stsClient successfully built");
+      AssumeRoleResponse response =
+          stsClient.assumeRole(
+              AssumeRoleRequest.builder()
+                  .roleSessionName("PolarisCredentialsSTS")
+                  .roleArn(storageConfig.getS3RoleArn())
+                  .policy(
+                      StorageUtil.policyString(
+                              storageConfig.getS3RoleArn(),
+                              allowListOperation,
+                              allowedReadLocations,
+                              allowedWriteLocations)
+                          .toJson())
+                  .durationSeconds(loadConfig(STORAGE_CREDENTIAL_DURATION_SECONDS))
+                  .build());
+
+      propertiesMap.put(PolarisCredentialProperty.AWS_KEY_ID, response.credentials().accessKeyId());
+      propertiesMap.put(
+          PolarisCredentialProperty.AWS_SECRET_KEY, response.credentials().secretAccessKey());
+      propertiesMap.put(PolarisCredentialProperty.AWS_TOKEN, response.credentials().sessionToken());
```

Comment:
Nit: using static import so that we can keep them in one line each.
```suggestion
      propertiesMap.put(AWS_KEY_ID, response.credentials().accessKeyId());
      propertiesMap.put(AWS_SECRET_KEY, response.credentials().secretAccessKey());
      propertiesMap.put(AWS_TOKEN, response.credentials().sessionToken());
```